src/login/oidc/redirectHandler/TokenSaver.ts

```diff
@@ -48,41 +47,39 @@ export default class TokenSaver implements ITokenSaver {
     idToken: string,
     accessToken?: string,
     refreshToken?: string
-  ): Promise<ISolidSession> {
-    const decoded = await this.joseUtility.decodeJWT(
-      // TODO this should actually be the id_vc of the token
-      accessToken as string
-    );
-    // TODO validate decoded token
-    // TODO extract the localUserId from state and put it in the session
-    const session = this.sessionCreator.create({
-      localUserId,
-      webId: decoded.sub as string,
-      neededAction: {
-        actionType: "inaction"
-      } as INeededInactionAction,
-      loggedIn: true
-    });
-    await this.storageUtility.setForUser(
-      session.localUserId,
-      "accessToken",
-      accessToken as string
-    );
-    await this.storageUtility.setForUser(
-      session.localUserId,
-      "idToken",
-      idToken as string
-    );
-    await this.storageUtility.setForUser(
-      session.localUserId,
-      "webId",
-      decoded.sub as string
-    );
-    await this.storageUtility.setForUser(
-      session.localUserId,
-      "refreshToken",
-      refreshToken as string
-    );
-    return session;
+  ): Promise<ISessionInfo> {
+    throw new Error("Not Implemented");
+    // const decoded = await this.joseUtility.decodeJWT(
```

Comment:
Ditto

Reply:
This is one of the pieces that need to be re-implemented to patch the security issues regarding token management. I'd leave the commented out code until it is replaced.